### PR TITLE
updated Building doc to Getting Started

### PR DESCRIPTION
**Pull request by jbloom22**
I kept things simple. Next round we should add a richer example. Note we'll update 1.5.2 to 1.6.2 once fix is in.

### CODE REVIEW

File: docs/GettingStarted.md [hunk 1]
```diff
@@ -0,0 +1,80 @@
+# Getting started
+
+All you'll a [Java 8 JDK](http://www.oracle.com/technetwork/java/javase/downloads/index.html) and the Hail source code: either clone the [Hail repository](https://github.com/broadinstitute/hail) using [Git](https://git-scm.com/) with
```
Comment (cseed) [inline docs/GettingStarted.md]: Not grammatical.

**Comment by tpoterba** [comment]
You can change it to 1.6.2.  The versioning fix is in

File: docs/GettingStarted.md [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+```
+git clone https://github.com/broadinstitute/hail.git
+```
+or download the source code directly from [here](https://github.com/broadinstitute/hail/archive/master.zip), unzip, and rename the enclosing `hail-master` directory to `hail`. The following commands are relative to this `hail` directory.
```
Comment (cseed) [inline docs/GettingStarted.md]: > The following commands are relative to this `hail` directory.

As written, this doesn't sound like it refers to the hail directory of the clone (which you don't even mention).  I'd prefer something like:

```
$ git clone ...
$ cd hail
```

The commands below are relative to the `hail` directory.

You can also download the source code directly from [here](...).

File: docs/GettingStarted.md [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+
+## Building and running Hail
+
+Hail may be built to run locally with all cores on a single machine or to run on a Spark cluster. Running locally is useful for getting started, analyzing or experimenting with small datasets, and Hail development.
```
Comment (cseed) [inline docs/GettingStarted.md]: Locally means on a single machine.  Pick one.  all cores is too much detail.

File: docs/GettingStarted.md [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+
+### Building and running Hail locally
+The single command
+```
+./gradlew installDist
```
Comment (cseed) [inline docs/GettingStarted.md]: Put `$ ` before shell commands.

File: docs/GettingStarted.md [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+```
+will populate `build/install` with an installation of Hail, including the executible `build/install/hail/bin/hail`. The initial build takes time as the build tool [Gradle](https://gradle.org/) installs and then downloads all Hail build dependencies, including the Scala compiler and Spark.
```
Comment (cseed) [inline docs/GettingStarted.md]: I would say `build/install/hail` is an installation of Hail.  executable (spelling).  I'd just say "dependencies".

File: docs/GettingStarted.md [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+
+Hail is compatible with Spark versions 1.5 and 1.6, and uses 1.5.2 by default. Set the Spark version to match that of the Spark cluster with
+```
+./gradlew -Dspark.version=VERSION
+```
+Then run
+```
+./gradlew shadowJar
+```
+to build `build/libs/hail-all-spark.jar`, which can be submitted using `spark-submit` (see [Spark documentation](http://spark.apache.org/docs/1.6.2/submitting-applications.html) for details).
```
Comment (cseed) [inline docs/GettingStarted.md]: I'd probably link to:

http://spark.apache.org/docs/1.6.2/cluster-overview.html

which gives slightly more context but links to submitting applications right at the top.

File: docs/GettingStarted.md [hunk 7]
```diff
@@ -0,0 +1,80 @@
...
+```
+If natives are not found, `hail.log` will contain the warnings
+```
+Failed to load implementation from: com.github.fommil.netlib.NativeSystemLAPACK
+Failed to load implementation from: com.github.fommil.netlib.NativeSystemBLAS
+```
+
+## Rebuilding and testing Hail
+
+Subsequent builds may at times fail due to artifacts from previous builds. To delete the `build` directory and start fresh, run
```
Comment (cseed) [inline docs/GettingStarted.md]: This is not strictly correct.  Clean does not delete artifacts.  I'd probably leave this out.  This should go in something like tips/tricks in the developer docs.

File: docs/GettingStarted.md [hunk 8]
```diff
@@ -0,0 +1,80 @@
...
+```
+Several Hail tests have additional dependencies:
+
+1. PLINK 1.9, [http://www.cog-genomics.org/plink2](http://www.cog-genomics.org/plink2)
+
+2. QCTOOL 1.4, [http://www.well.ox.ac.uk/~gav/qctool](http://www.well.ox.ac.uk/~gav/qctool)
+
+3. R 3.3.1, [http://www.r-project.org/](http://www.r-project.org/) with packages `jsonlite`, `SKAT`, and `logistf`, which depends on `mice` and `Rcpp`.
+
+Other recent minor version of QCTOOL and R should suffice, but PLINK 1.0 will not.
```
Comment (cseed) [inline docs/GettingStarted.md]: Not grammatical.

File: docs/GettingStarted.md [hunk 9]
```diff
@@ -0,0 +1,80 @@
...
+```
+./build/install/hail/bin/hail read -i ~/sample.vds variantqc -o ~/variantqc.tsv sampleqc -o ~/sampleqc.tsv
+```
+To run Hail using `hail` in place of `./build/install/hail/bin/hail`, add this `bin` directory to your path. Note that during each run Hail writes a `hail.log` file in the current directory; this is useful to developers for debugging.
+
+### Building a Spark application JAR
+
+Hail is compatible with Spark versions 1.5 and 1.6, and uses 1.5.2 by default. Set the Spark version to match that of the Spark cluster with
+```
+./gradlew -Dspark.version=VERSION
```
Comment (cseed) [inline docs/GettingStarted.md]: No.  You have to set the version when you build the shadowJar.

File: docs/GettingStarted.md [hunk 10]
```diff
@@ -0,0 +1,80 @@
...
+```
+Then run
+```
+./gradlew shadowJar
+```
+to build `build/libs/hail-all-spark.jar`, which can be submitted using `spark-submit` (see [Spark documentation](http://spark.apache.org/docs/1.6.2/submitting-applications.html) for details).
+
+### BLAS and LAPACK
+
+Hail uses the Scala numerical processing library [Breeze](http://github.com/scalanlp/breeze), which in turn uses the [netlib-java](http://github.com/fommil/netlib-java) to take advantage of native [BLAS](http://www.netlib.org/blas/) and [LAPACK](http://www.netlib.org/lapack/) routines. These routines are available by default on OS X and Windows. On Linux, they must be explicitly enabled. On Ubuntu 14.04, run
```
Comment (cseed) [inline docs/GettingStarted.md]: Detail overload.  Just say "Hail uses BLAS and LAPACK optimized linear algebra [or numeric processing] libraries."

Say installed, not enabled.

BLAS and LAPACK are available by default on Windows?  To my knowledge, nobody has tried to build or run Hail on Windows.  I think you shouldn't mention it.

File: docs/GettingStarted.md [hunk 11]
```diff
@@ -0,0 +1,80 @@
...
+```
+./gradlew test
+```
+Several Hail tests have additional dependencies:
+
+1. PLINK 1.9, [http://www.cog-genomics.org/plink2](http://www.cog-genomics.org/plink2)
+
+2. QCTOOL 1.4, [http://www.well.ox.ac.uk/~gav/qctool](http://www.well.ox.ac.uk/~gav/qctool)
+
+3. R 3.3.1, [http://www.r-project.org/](http://www.r-project.org/) with packages `jsonlite`, `SKAT`, and `logistf`, which depends on `mice` and `Rcpp`.
```
Comment (cseed) [inline docs/GettingStarted.md]: I'd bullet rather than number these.  They're not naturally ordered.

**Comment by jbloom22** [comment]
back to you

File: docs/GettingStarted.md [hunk 12]
```diff
@@ -0,0 +1,76 @@
+# Getting started
+
+All you'll need is a [Java 8 JDK](http://www.oracle.com/technetwork/java/javase/downloads/index.html) and the Hail source code. To clone the [Hail repository](https://github.com/broadinstitute/hail) using [Git](https://git-scm.com/), run
```
Comment (cseed) [inline docs/GettingStarted.md]: I'd say "the Java 8 JDK".

File: docs/GettingStarted.md [hunk 13]
```diff
@@ -0,0 +1,76 @@
...
+## Building and running Hail
+
+Hail may be built to run locally or on a Spark cluster. Running locally is useful for getting started, analyzing or experimenting with small datasets, and Hail development.
+
+### Locally
+The single command
+```
+$ ./gradlew installDist
+```
+installs Hail at `build/install/hail`. The initial build takes time as the build tool [Gradle](https://gradle.org/) installs and then downloads all Hail dependencies. The executable is `build/install/hail/bin/hail` (to run using `hail` add this `bin` to your path).
```
Comment (cseed) [inline docs/GettingStarted.md]: "installs and then downloads" sounds strange.  The gradle wrapper downloads gradle.  Again, too much detail.  I'd just drop "installs".

Comment (cseed) [inline docs/GettingStarted.md]: "this bin" sounds strange.  Maybe "add `build/install/hail/bin` to your path."

File: docs/GettingStarted.md [hunk 14]
```diff
@@ -0,0 +1,76 @@
...
+```
+$ ./build/install/hail/bin/hail read -i ~/sample.vds variantqc -o ~/variantqc.tsv sampleqc -o ~/sampleqc.tsv
+```
+Note that during each run Hail writes a `hail.log` file in the current directory; this is useful to developers for debugging.
+
+### Spark cluster
+
+Hail is compatible with Spark versions 1.5 and 1.6, and uses 1.6.2 by default. Run
+```
+$ ./gradlew shadowJar -Dspark.version=VERSION
```
Comment (cseed) [inline docs/GettingStarted.md]: Put the -D before the shadowJar.

File: docs/GettingStarted.md [hunk 15]
```diff
@@ -0,0 +1,76 @@
...
+```
+using the version of Spark on your cluster. The resulting JAR `build/libs/hail-all-spark.jar` can be submitted using `spark-submit`. See [Spark documentation](http://spark.apache.org/docs/1.6.2/cluster-overview.html) for details.
+
+### BLAS and LAPACK
+
+Hail uses BLAS and LAPACK-optimized linear algebra libraries. On Linux, these must be explicitly installed. On Ubuntu 14.04, run
```
Comment (cseed) [inline docs/GettingStarted.md]: No dash.

File: docs/GettingStarted.md [hunk 16]
```diff
@@ -0,0 +1,76 @@
...
+$ apt-get install libatlas-base-dev
+```
+If natives are not found, `hail.log` will contain the warnings
+```
+Failed to load implementation from: com.github.fommil.netlib.NativeSystemLAPACK
+Failed to load implementation from: com.github.fommil.netlib.NativeSystemBLAS
+```
+See [netlib-java](http://github.com/fommil/netlib-java) for more information.
+
+## Testing Hail
```
Comment (cseed) [inline docs/GettingStarted.md]: I like the sound of "Running the Tests" better.  Presumably it is already tested.  The user is just testing their setup/installation.

File: docs/GettingStarted.md [hunk 17]
```diff
@@ -0,0 +1,76 @@
...
+
+To execute all Hail tests, run
+```
+$ ./gradlew test
+```
+Several Hail tests have additional dependencies:
```
Comment (cseed) [inline docs/GettingStarted.md]: Put the dependencies first.

**Pull request merged**